src/test/scala/fi/vm/sade/hakurekisteri/integration/ytl/YtlIntegrationSpec.scala

```diff
@@ -517,6 +519,41 @@ class YtlIntegrationSpec extends FlatSpec with BeforeAndAfterEach with BeforeAnd
     Mockito.verify(failureEmailSenderMock, Mockito.times(1)).sendFailureEmail(mockito.ArgumentMatchers.any(classOf[String]))
   }
 
+  it should "fail if ytl fetch returns throwables" in new UseYtlKokelasPersister with UseYtlIntegration {
+    Mockito.when(hakemusService.hetuAndPersonOidForHaku(activeHakuOid)).thenReturn(Future.successful(tenEntries))
+
+    private val ytlHttpClientThatReturnsThrowables: YtlHttpFetch = mock[YtlHttpFetch]
+    private val lefts = Seq(Left(new RuntimeException("mocked failure")))
+    Mockito.when(ytlHttpClientThatReturnsThrowables.fetch(mockito.ArgumentMatchers.any(classOf[String]), mockito.ArgumentMatchers.any(classOf[Seq[String]])))
+      .thenReturn(lefts.toIterator)
+
+    val realKokelasPersister = createTestYtlKokelasPersister()
+    val ytlIntegration = createTestYtlIntegration(realKokelasPersister)
+
+    ytlIntegration.syncAll(failureEmailSender = failureEmailSenderMock)
+
+    Thread.sleep(11000)
```

Comment:
Vähän inhottavia nämä kovakoodatut suhteellisen pitkät odottelut. Näitä voisi varmaan ainakin lyhentää, jos ei kokonaan eroon pääse.